bin/api_diff.ml

```diff
@@ -1,24 +1,31 @@
-let run (`Ref_cmi reference) (`Current_cmi current) =
+let run (`Main_module main_module) (`Ref_cmi reference) (`Current_cmi current) =
   let open CCResult.Infix in
   let* reference_sig, current_sig, module_name =
-    if Sys.is_directory reference && Sys.is_directory current then
-      let+ reference_sig = Api_watch.Library.load reference
-      and+ current_sig = Api_watch.Library.load current in
-      let module_name =
-        let rec find_library_name path =
-          let parent = Filename.dirname path in
-          if Filename.basename parent = "_build" then
-            Filename.basename (Filename.dirname parent)
-          else if Filename.basename parent = "_opam" then Filename.basename path
-          else find_library_name parent
+    match
+      (Sys.is_directory reference, Sys.is_directory current, main_module)
+    with
+    | true, true, Some main_module ->
+        let+ reference_sig = Api_watch.Library.load reference
+        and+ current_sig = Api_watch.Library.load current in
+        let module_name = String.capitalize_ascii main_module in
+        (reference_sig, current_sig, module_name)
+    | false, false, main_module ->
+        let () =
+          match main_module with
+          | None -> ()
+          | Some _ ->
+              Printf.eprintf
+                "%s: --main-module ignored when diffing single .cmi files"
+                Sys.executable_name
```

Comment:
I think we can also add a test for this (when both files are simple .cmi and a `--main-module` is also provided to the terminal) as a part of the cram tests.

Reply:
You're right, I'll add a test!